### PR TITLE
mpw designs for CI

### PR DESCRIPTION
**Pull request by vijayank88**
Signed-off-by: vijayank88 <paruthi143@gmail.com>
Fixed2Float MPW design added

### CODE REVIEW

**Review by vvbandeira — DISMISSED** [review]

File: scripts/setup-flow.sh [hunk 1]
```diff
@@ -1,4 +1,8 @@
 # copy config files to design dir
-rsync --archive designs/${design}/. ${design_dir}
+if [ -z "$NICKNAME" ]; then
```
Comment (vijayank88) [inline scripts/setup-flow.sh:2]: @vvbandeira  
Locally tested its working fine with command: ./scripts/run-design.sh wb_host mbist_ctrl
In CI its got failed during rsync. Can you please review this and why its got failed.
Thanks

Comment (vvbandeira) [reply scripts/setup-flow.sh:2]: Why not use `rsync --archive designs/mbist_ctrl/. ${design_dir}` and remove the if statement?

**Review by vvbandeira — DISMISSED** [review]

**Comment by vijayank88** [comment]
@vvbandeira  
shall I push other designs as well in same branch or I need to create new branch again?

File: scripts/flexbex_core/config.tcl [hunk 2]
```diff
@@ -0,0 +1,6 @@
+cd [file dirname [file normalize [info script]]]
+set ::env(CARAVEL_ROOT) "[pwd]/caravel"
+set ::env(DESIGN_IS_CORE) 1
+#set ::env(PL_RESIZER_HOLD_SLACK_MARGIN) 0.3
+#set ::env(GLB_RESIZER_HOLD_SLACK_MARGIN) 0.5
```
Comment (vvbandeira) [inline scripts/flexbex_core/config.tcl:5]: remove

File: scripts/flexbex_core/setup-flow.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# copy config files to design dir
+rsync --archive designs/soric_project/. ${design_dir}
+cp -f ./scripts/${design}/config.tcl ${design_dir}
+sed -i '/::env(DESIGN_IS_CORE)/d' ${design_dir}/openlane/flexbex_core/config.tcl
+sed -i '/::env(ROUTING_CORES)/d' ${design_dir}/openlane/flexbex_core/config.tcl
+sed -i '/::env(GLB_RT_MAXLAYER)/d' ${design_dir}/openlane/flexbex_core/config.tcl
+sed -i '/::env(SYNTH_STRATEGY)/d' ${design_dir}/openlane/flexbex_core/config.tcl
+#sed -i '/::env(PL_RESIZER_HOLD_SLACK_MARGIN)/d' ${design_dir}/openlane/flexbex_core/config.tcl
+#sed -i '/::env(GLB_RESIZER_HOLD_SLACK_MARGIN)/d' ${design_dir}/openlane/flexbex_core/config.tcl
```
Comment (vvbandeira) [inline scripts/flexbex_core/setup-flow.sh:9]: remove

**Review by vvbandeira — CHANGES_REQUESTED** [review]
See comments above and also remove the following temp files:
scripts/flexbex_core/.config.tcl.swp
scripts/flexbex_core/.setup-flow.sh.swp

**Comment by vijayank88** [comment]
@vvbandeira 
During CI runs any one of designs get failed for `python3 -m pip install --user pyyaml click` for this jacaranda8 failed.

OL tage updated to [2022.02.21_02.33.47](https://github.com/The-OpenROAD-Project/OpenLane/releases/tag/2022.02.21_02.33.47) tag gives Magic DRC violation for `FCNet_prj` but 2022.02.16_01.55.24 has no magic DRC violations.
@dralabeing FYI

**Comment by vijayank88** [comment]
[bd105fd](https://github.com/The-OpenROAD-Project/OpenLane-MPW-CI/pull/4/commits/bd105fd97ab4f36888d0bfddf12998c5ae1cb976) this commit got failed with hold violations for soric_soc

With previous commit  [53b75ad](https://github.com/The-OpenROAD-Project/OpenLane-MPW-CI/pull/4/commits/53b75ad95fa53c6184038d5f0b97a12505f7de6a) soric_soc completed successfully without any violations.

**Comment by vvbandeira** [comment]
```
set ::env(QUIT_ON_MAGIC_DRC) 0
set ::env(QUIT_ON_HOLD_VIOLATIONS) 0
```
I see this for two designs. This will allow DRCs or hold violations, is this the behaviour we want for the CI?
@dralabeing can you comment?

**Comment by dralabeing** [comment]
We should catch DRC and hold violations and hence the CI should fail. These env variables should be set to 1 since these are serious errors. If successive releases give DRC and timing violations then @vijayank88  file a GH issue against OL with the tag specified.

**Comment by vijayank88** [comment]
@vvbandeira  With this commit added only design with clean pass and no manual changes with configuration file.
Other designs scripts added, once GitHub issue got resolved, will check with latest OL tag and will be added to `Jenkinsfile` for CI

File: scripts/crypto_core/setup-flow.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# copy config files to design dir
+rsync --archive designs/soric_project/. ${design_dir}
+cp -f ./scripts/${design}/config.tcl ${design_dir}
+sed -i '/::env(DESIGN_IS_CORE)/d' ${design_dir}/openlane/crypto_core/config.tcl
+sed -i '/::env(ROUTING_CORES)/d' ${design_dir}/openlane/crypto_core/config.tcl
+sed -i '/::env(GLB_RT_MAXLAYER)/d' ${design_dir}/openlane/crypto_core/config.tcl
+#sed -i '/::env(SYNTH_STRATEGY)/d' ${design_dir}/openlane/crypto_core/config.tcl
```
Comment (vvbandeira) [inline scripts/crypto_core/setup-flow.sh:7]: just remove it

**Review by vvbandeira — CHANGES_REQUESTED** [review]

**Review by vvbandeira — APPROVED** [review]

**Pull request merged**